src/mantarray_desktop_app/mc_simulator.py

```diff
@@ -131,46 +145,80 @@ def _send_data_packet(
 
     def _commands_for_each_run_iteration(self) -> None:
         self._handle_test_comm()
+        # if _reboot_time_secs is not None, this means the simulator is in a "reboot" phase
+        if self._reboot_time_secs is not None:
+            secs_since_reboot = _get_secs_since_reboot_command(self._reboot_time_secs)
+            # if secs_since_reboot is less than the reboot duration, simulator is still in the 'reboot' phase. Commands from PC will be discared and status beacons will not be sent
+            if secs_since_reboot < 5:
+                self._discard_comm_from_pc()
+                return
+            self._handle_reboot_completion()
         self._handle_comm_from_pc()
         self._handle_status_beacon()
 
+    def _handle_reboot_completion(self) -> None:
+        self._reset_start_time()
+        self._reboot_time_secs = None
+        self._reset_status_code_bits()
+        self._send_status_beacon(truncate=False)
+
+    def _discard_comm_from_pc(self) -> None:
+        try:
+            self._input_queue.get_nowait()
```

Comment:
if we do need to discard all commands from the PC during a reboot, would this accomplish that? this is only discarding a single command each cycle.  would it be a more accurate representation of the Mantarray to just drain the queue as part of `_handle_reboot_completion`?